src/common/common.cpp

```diff
@@ -385,6 +385,25 @@ std::wstring get_resource_string(UINT resource_id, HINSTANCE instance, const wch
   }
 }
 
+// function to return the string as a to a wchar_t* (to enable localization in cases where the return type did not accept wstring)
+wchar_t* get_resource_string_wchar(UINT resource_id, HINSTANCE instance)
+{
+    wchar_t* text_ptr;
+    unsigned int length = LoadStringW(instance, resource_id, reinterpret_cast<wchar_t*>(&text_ptr), 0);
+    std::wstring res_string = { *reinterpret_cast<wchar_t**>(&text_ptr), length };
+    length++;
+
+    if (length > 1)
+    {
+        wchar_t* tmp_res_ptr;
+        tmp_res_ptr = new wchar_t[length];
+        wmemcpy(tmp_res_ptr, res_string.c_str(), length);
```

Comment:
Why not just return the read-only pointer `text_ptr`?

Reply:
Not applicable anymore. Modified to use std::wstring instead.